dapps/digital-twin-detail/src/components/DigitalTwinDetail.vue

```diff
@@ -58,40 +58,107 @@
                         icon="mdi mdi-dots-vertical"
                       />
                     </template>
-                    <b-dropdown-item href="#">TODO Clone</b-dropdown-item>
+                    <b-dropdown-item
+                      @click="duplicateTwin()">
+                      {{ '_twin-detail.data.context-menu.duplicate-twin' | translate }}
+                    </b-dropdown-item>
                     <b-dropdown-item
                       @click="exportTwinTemplate()"
                     >{{ '_twin-detail.data.context-menu.export-template' | translate }}</b-dropdown-item>
                   </b-dropdown>
 
-                  <evan-modal ref="exportModal">
-                    <template v-slot:header>
-                      <h5 class="modal-title">
-                        {{ `_twin-detail.data.context-menu.export-template` | translate }}
-                      </h5>
-                    </template>
-
-                    <template v-slot:body>
-                      <evan-loading v-if="exporting" />
+                  <evan-swipe-panel
+                    @init="exportModal = $event"
+                    alignment="right"
+                    class="light"
+                    :title="'_twin-detail.data.context-menu.export-template' | translate"
+                    showBackdrop="true"
+                    type="default">
+                    <div class="d-flex h-100 align-items-center justify-content-center">
+                      <div v-if="exporting">
+                        <evan-loading />
+                        <div class="mt-3 text-center">
+                          <h4>
+                            {{ `_twin-detail.data.context-menu.exporting-twin` | translate }}
+                          </h4>
+                        </div>
+                      </div>
                       <evan-success v-else />
+                    </div>
+
+                    <template slot="footer">
+                      <div class="d-flex">
+                        <evan-button
+                          type="secondary"
+                          :label="$t('_evan.cancel')"
+                          @click="exportModal.hide()"
+                          class="mr-3"
+                        />
+                        <span class="mx-auto" />
+                        <evan-button
+                          type="primary"
+                          :disabled="exporting"
+                          :label="$t('_twin-detail.data.context-menu.download-template')"
```

Comment:
translation function or filter? We should use only one style in .vue template

Reply:
I would prefer to use translate pipe when it's possible.